zeebe/atomix/cluster/src/main/java/io/atomix/cluster/messaging/impl/NettyMessagingService.java

```diff
@@ -508,6 +532,36 @@ private void initTransport() {
     }
   }
 
+  private X509Certificate[] getCertificateChain(final File pkcs12File, final String password)
+      throws CertificateException, KeyStoreException, IOException, NoSuchAlgorithmException {
+    final var keyStore = getKeyStore(pkcs12File, password);
+
+    final String alias = keyStore.aliases().nextElement();
```

Comment:
When an entry is added to a pkcs12 file you must define an alias,key, and certchain (with option key password). This will associate the alias with the key and the cert chain. so for one entry there will be one alias the intermediate certs will have the same alias as the leaf/root.

`
setKeyEntry(alias, key, keyPassword, certChain)
chain = keyStore.getCertificateChain(alias)
chain[0] = leaf
chain[1] = intermediate
chain[2] = root

key = keyStore.getKey(alias, keyPassword)
`

The above is how things work (function names are not 100% accurate).